### ISSUE TITLE
Load _asdf object early, validate later after FITS keywords are mapped to the tree

### ISSUE THREAD

**Issue by jdavies-st** [spacetelescope/stdatamodels#20]
PR #19 implemented a band-aid solution for reading FITS files.  @eslavich and I discussed this a bit, and have some concerns:

Because it populates `DataModel._asdf` with a dummy `AsdfFile` it is possible that a bunch of validation could occur using a different version of ASDF standard than was is contained in the FITS file being read.

It might be better to just load `hdulist["ASDF"]` using `AsdfInFits` early, and use it as the context, so that there's no mismatch, just in case file being read in uses a different ASDF standard version, or is different in some other way.

Currently, the function that populates the `DataModel` from a `HDUList` returns the `AsdfInFits` instance, and only then does it get assigned to `_asdf`.

Calls to `validate.value_change` are happening in `fits_support._load_from_schema`.  We're calling that method to determine if the fits value is good enough to be assigned, and skipping it if not.  Do we need to do this check here?  Or is validating at the end good enough?

Finally, it seems that if the FITS hash matches, no validation occurs, not even against the schema of the final tree.  Not sure if this is intentional or not.

**Comment by braingram** [comment]
tldr; this issue still applies even though the conditions have changed slightly

Although some things have changed since this issue was opened (there is no longer an `AsdfInFits`) the general structure and concerns are the same (now).

`_asdf` is initialized with a default `AsdfFile` instance:
https://github.com/spacetelescope/stdatamodels/blob/192df47c1390ef8bac2ab115531606969980c1ec/src/stdatamodels/model_base.py#L178-L181
prior to the call to `from_fits`:
https://github.com/spacetelescope/stdatamodels/blob/192df47c1390ef8bac2ab115531606969980c1ec/src/stdatamodels/model_base.py#L236
which as stated above uses the `_asdf` instance for validation during calls to `value_change` for assignment validation (under some conditions).

I believe this is only an issue for a FITS file (which might currently be the only condition where `value_change` might be called). Since the `AsdfFile` instance for the read file is created here:
https://github.com/spacetelescope/stdatamodels/blob/192df47c1390ef8bac2ab115531606969980c1ec/src/stdatamodels/fits_support.py#L725
before it's used:
https://github.com/spacetelescope/stdatamodels/blob/192df47c1390ef8bac2ab115531606969980c1ec/src/stdatamodels/fits_support.py#L730
It looks doable to use the read `AsdfFile` instance before any calls to `value_change`.
